src/lsp/cobol_parser/expect/gen_expect.ml

```diff
@@ -312,16 +324,30 @@ let default_nonterminal_value (type a): a Menhir.nonterminal -> a = function
      guessed the type of a nonterinal, either change the name
      or add a @default attribute on the faulty nonterminal *)
 |};
-  Nonterminal.fold (fun nt acc -> begin
-        match best_guess_default_value nt with
-        | None -> acc
-        | Some default -> ([nt], default)::acc
-      end) []
+  Nonterminal.fold begin fun nt acc ->
+    match best_guess_default_value nt with
+    | None -> acc
+    | Some default -> ([nt], default)::acc
+  end []
   |> pp_match_cases ppf
     Fmt.(using Nonterminal.mangled_name (any "N_" ++ string))
     Fmt.string
     "raise Not_found"
 
+let ensure_all_nullable_nonterminal_have_default () =
+  let nt_without_default = Nonterminal.fold (fun nt acc -> begin
+        if Nonterminal.nullable nt
+        && best_guess_default_value nt == None
+        then nt::acc else acc
+      end) [] in
+  match List.hd nt_without_default with
+  | nt ->
+      let error_msg = Pretty.to_string
+        "Unable to find default value for the nonterminal '%s', please provide a @default annotation"
```

Comment:
```suggestion
        "Unable to find a default value for nonterminal '%s'; please provide a @@default annotation"
```

Reply:
You can also `Pretty.failwith "…"` directly (no need for the extra binding)

Reply:
Should also probably all go in a `Nonterminal.iter begin fun nt -> …`. That'd be much cleaner than with the preliminary `fold`. (cf other comment)